Cython/Debugger/Cygdb.py

```diff
@@ -45,17 +45,22 @@ def make_command_file(path_to_debug_info, prefix_code='', no_import=False):
             set print pretty on
 
             python
-            # Activate virtualenv, if we were launched from one
-            import os
-            virtualenv = os.getenv('VIRTUAL_ENV')
-            if virtualenv:
-                path_to_activate_this_py = os.path.join(virtualenv, 'bin', 'activate_this.py')
-                print("gdb command file: Activating virtualenv: %s; path_to_activate_this_py: %s" % (
-                    virtualenv, path_to_activate_this_py))
-                with open(path_to_activate_this_py) as f:
-                    exec(f.read(), dict(__file__=path_to_activate_this_py))
-
-            from Cython.Debugger import libcython, libpython
+            from traceback import print_exc
+            try:
+                # Activate virtualenv, if we were launched from one
+                import os
+                virtualenv = os.getenv('VIRTUAL_ENV')
+                if virtualenv:
+                    path_to_activate_this_py = os.path.join(virtualenv, 'bin', 'activate_this.py')
+                    print("gdb command file: Activating virtualenv: %s; path_to_activate_this_py: %s" % (
+                        virtualenv, path_to_activate_this_py))
+                    with open(path_to_activate_this_py) as f:
+                        exec(f.read(), dict(__file__=path_to_activate_this_py))
+                from Cython.Debugger import libcython, libpython
+            except Exception as ex:
+                print("There was an error in Python code originating from the file ''' + str(__file__) + '''")
```

Comment:
Is `__file__` really the right file to report here? Or should it point at the virtualenv (if any) that it just tried to initialise?

Reply:
I think that __file__ is correct I installed it in a virtualenv and it printed:
/home/volker/Documents/venv/lib/python3.8/site-packages/Cython/Debugger/Cygdb.py
which is exactly what we want.